languages/cpp/reference/exercise-concepts/reverse-string.md

```diff
@@ -0,0 +1,77 @@
+# Reverse String
+
+## Task
+
+Reverse a string
+
+For example:
+input: "cool"
+output: "looc"
+
+
+## Example
+
+`reverse_string.h`
+
+```cpp
+#if !defined(REVERSE_STRING_H)
+#define REVERSE_STRING_H
+
+#include <string>
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text);
+
+}
+
+#endif
+```
+
+`reverse_string.cpp`
+```cpp
+#include "reverse_string.h"
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text)
+{
+    return std::string(text.rbegin(), text.rend());
+}
+
+}
+```
+
+## Concepts
+
+- .h file: the tests include `"reverse_string.h"`
+- .cpp file:  the most common approach is to use a `.cpp` for the definitions
+- namespaces: the function is member of a namespace
+- functions: used as the main entry point for the exercise
+- function arguments: the function has two arguments
+- strings: input is a string (or maybe string_view)
+- return values: the result is returned by the function
+- pass by reference to const: avoid making copy of input
+- iterators: view input as a range
+
+
+## Alternative approaches
+
```

Comment:
`std::reverse()` from algorithm library looks like good alternative approach.

Reply:
In this case, i do not think so. It requires additional copy to be constructed before you can call it. Also, according to the compiler explorer, it generates more code on gcc-trunk.

Reply:
i fully agree. it's not the most **effective** way to solve it. But there are a lot of approved solutions on `exercism` used this approach.

Reply:
I agree with @marvelou-s here, alternative approaches should not take into consideration efficiency too much.
If someone submitted a solution that's not the most efficient we should take it into consideration anyway.

Reply:
I see your point, I'll add that to the list of alternative approaches.